### PR TITLE
Add alias and chainIds to manifest entry for context

### LINKED ISSUE #596: Add more context about child chains in manifest tree during build
At the moment the only chain with all of the chainIDs is the root chain. Further down the manifest tree the child chains have only the chain IDs for their children. They also need the chain IDs for chains they are joined to

It would also be beneficial if the child chains knew their own alias, for ease of processing in the root covenant.

During the build step
- add chain IDs of joined chains to context for chain entries in the manifest tree
- add the alias of the chain to context for chain entries in the manifest tree

### PR DESCRIPTION
**Pull request by nicholeous**
Closes #596 

Adds the alias of the manifest entry and the chain IDs of all children and joined chains to all manifest entries for context while processing in the root covenant.

Also a test.

### CODE REVIEW

File: packages/interbit/src/manifest/createManifestTree.js [hunk 1]
```diff
@@ -34,8 +34,12 @@ const createManifestTree = (interbitConfig, manifest) => {
     minimumJoinconfig
   )
 
+  const chainIds = mergeJoinAndChildChainIds(joins, chains, manifest)
+
   const manifestEntry = {
+    alias: ROOT_CHAIN_ALIAS,
     chainId,
+    chainIds,
```
Comment (djlmills) [inline packages/interbit/src/manifest/createManifestTree.js]: `chainAlias`, `chainId` and `covenant` are obvious, but `chainIds` and `covenants` less so. What chain IDs and covenants are being referred to? Could we have more descriptive labels? I would think `childChainIds`, `joinedChainIds` and `childChainCovenants` would be more explicit, and `childChainAliases` and `joinedChainAliases` would be really helpful to a reader.

Comment (karikushi) [reply packages/interbit/src/manifest/createManifestTree.js]: I second this. `chainId` and `chainIds` are also too easy to mix up.

Comment (nicholeous) [reply packages/interbit/src/manifest/createManifestTree.js]: The chainIds being referred to are all joined and child chain IDs. I am at a loss for a name that isn't something extreme like `childAndJoinedChainIds`. I may just replace the aliases with the chainIDs and forget that chainIds property altogether. 

Going to sleep on it but if you can think of a better name in the meantime lmk :)

Comment (nicholeous) [reply packages/interbit/src/manifest/createManifestTree.js]: They are glommed together as a reference map of all the chainIds you may interact with within a covenant so they can be used from anywhere with just an alias. Per your (in person) suggestion I think I will go with referenceChainIds and referenceCovenants

**Pull request merged**